master/buildbot/changes/gitpoller.py

```diff
@@ -159,8 +170,16 @@ def poll(self):
             '+%s:%s' % (self._removeHeads(branch), self._trackerBranch(branch))
             for branch in branches
         ]
-        yield self._dovccmd('fetch',
-                            [self.repourl] + refspecs, path=self.workdir)
+
+        while True:
+            try:
+                yield self._dovccmd('fetch', [self.repourl] + refspecs,
+                                    path=self.workdir)
+            except GitError, e:
+                log.msg(e)
+                asyncSleep(self.pollInterval)
```

Comment:
I actually yielded this line before. But then, I messed up with a force push, did it whole PR all again from memory and forgot the `yield`. I'll fix it asap.